### PR TITLE
Use errno.EISDIR and errno.EACCESS instead of hardcoding values

### PR DESCRIPTION
**Pull request by dktrkranz**

### CODE REVIEW

**Comment by arsenetar** [comment]
Windows tests show failures with this change.  I think both codes need to be kept.  The Linux one can be replaced with `EISDIR`.

**Comment by dktrkranz** [comment]
Indeed Windows seems to rely on EACCES, added that as well.

File: core/results.py [hunk 1]
```diff
@@ -375,9 +376,8 @@ def do_write(outfile):
         try:
             do_write(outfile)
         except OSError as e:
-            # If our OSError is because dest is already a directory, we want to handle that. 21 is
-            # the code we get on OS X and Linux, 13 is what we get on Windows.
-            if e.errno in {21, 13}:
+            # If our OSError is because dest is already a directory, we want to handle that.
+            if e.errno in (EISDIR, EACCES):
```
Comment (arsenetar) [inline core/results.py:381]: I do think maybe leaving the note about how different errors here are due to the different OSes would be helpful in the future. Same in the other file as well.

Comment (dktrkranz) [reply core/results.py:381]: Comments refactored as per suggestion

**Pull request merged**